hashes/md5.py

```diff
@@ -1,127 +1,329 @@
-import math
+"""
+The MD5 algorithm is a hash function that's commonly used as a checksum to
+detect data corruption. The algorithm works by processing a given message in
+blocks of 512 bits, padding the message as needed. It uses the blocks to operate
+a 128-bit state and performs a total of 64 such operations. Note that all values
+are little-endian, so inputs are converted as needed.
 
+Although MD5 was used as a cryptographic hash function in the past, it's since
+been cracked, so it shouldn't be used for security purposes.
 
-def rearrange(bit_string_32):
-    """[summary]
-    Regroups the given binary string.
+For more info, see https://en.wikipedia.org/wiki/MD5
+"""
+
+from collections.abc import Generator
+from math import sin
```

Comment:
Please use https://docs.python.org/3/library/struct.html to do the endian conversations or at least use `struct` in the doctests for  `to_little_endian()`.

Reply:
Admittedly I'm not that familiar with the `struct` module, but I'm not sure how to make it work with the `to_little_endian()` function.

To me, the problem is that the original function `rearrange()` doesn't _actually_ convert strings to little-endian. Instead, it treats 32-char string inputs as if they were _32-bit bit strings_, with each char as a single "bit". It then restructures the input in an "little-endian fashion": the 8 least significant "bits" come first, followed by the 8 next least significant "bits", etc. Thus it looks little-endian if you squint hard enough.

Since the inputs to `rearrange()`/`to_little_endian()` are being restructured in units far larger than a byte, I'm not sure if the `struct` module would work here, unless I'm misunderstanding how the module works.